### PR TITLE
Bump dependency restriction to enable use with python 3.13

### PR DESCRIPTION
**Pull request by kbwestfall**
As titled.

I'll try to set off a cloud dev-suite run forcing it to use python 3.13.

### CODE REVIEW

**Comment by kbwestfall** [comment]
Tests pass on python 3.13:
```
Test Summary
--------------------------------------------------------
--- PYTEST PYPEIT UNIT TESTS PASSED  257 passed, 6569 warnings in 570.38s (0:09:30) ---
--- PYTEST UNIT TESTS PASSED  152 passed, 2670 warnings in 1047.79s (0:17:27) ---
--- PYTEST VET TESTS PASSED  63 passed, 106684 warnings in 5983.77s (1:39:43) ---
--- PYPEIT DEVELOPMENT SUITE PASSED 248/248 TESTS  ---
Testing Started at 2024-12-18T00:48:53.800403
Testing Completed at 2024-12-18T20:51:13.380914
Total Time: 20:02:19.580511
```

**Review by rcooke-ast — APPROVED** [review]

File: pypeit/core/qa.py [hunk 1]
```diff
@@ -132,22 +130,17 @@ def get_dimen(x, maxp=25):
...
-    """
-    tstamp = datetime.datetime.today().strftime('%Y-%m-%d-T%Hh%Mm%Ss')
-    try:
-        user = getpass.getuser()
-    except ModuleNotFoundError:
-        # there appears to be a bug in getpass in windows systems where the pwd module doesn't load
-        user = os.getlogin()
-    # Return
-    return '{:s}_{:s}'.format(user, tstamp)
+# THIS IS ONLY USED BY DEPRECATED FUNCTIONS
```
Comment (rcooke-ast) [inline pypeit/core/qa.py]: Should we just deprecate?

Comment (kbwestfall) [reply pypeit/core/qa.py]: Done!

**Review by debora-pe — APPROVED** [review]
Thanks @kbwestfall

**Pull request merged**